tests/vmi_tpm_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package tests_test
+
+import (
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/tests/framework/checks"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/console"
+)
+
+var _ = Describe("[sig-compute]vTPM", func() {
+	var virtClient kubecli.KubevirtClient
+	var err error
+
+	BeforeEach(func() {
+		virtClient, err = kubecli.GetKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	Context("with TPM VMI option enabled", func() {
+		It("should expose a functional emulated TPM which persists across migrations", func() {
+			By("Creating a VMI with TPM enabled")
+			vmi := tests.NewRandomFedoraVMI()
+			vmi.Spec.Domain.Devices.TPM = &v1.TPMDevice{}
+			vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+
+			By("Logging in as root")
+			err = console.LoginToFedora(vmi)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("Ensuring a TPM device is present")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "ls /dev/tpm*\n"},
+				&expect.BExp{R: "/dev/tpm0"},
+			}, 300)).To(Succeed(), "Could not find a TPM device")
+
+			By("Ensuring the TPM device is functional")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: "tpm2_pcrread sha256:15\n"},
+				&expect.BExp{R: "0x0000000000000000000000000000000000000000000000000000000000000000"},
+				&expect.BSnd{S: "tpm2_pcrextend 15:sha256=54d626e08c1c802b305dad30b7e54a82f102390cc92c7d4db112048935236e9c && echo 'do''ne'\n"},
+				&expect.BExp{R: "done"},
+				&expect.BSnd{S: "tpm2_pcrread sha256:15\n"},
+				//printf '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x54\xd6\x26\xe0\x8c\x1c\x80\x2b\x30\x5d\xad\x30\xb7\xe5\x4a\x82\xf1\x02\x39\x0c\xc9\x2c\x7d\x4d\xb1\x12\x04\x89\x35\x23\x6e\x9c' | sha256sum
```

Comment:
@jean-edouard I think its a leftover

Reply:
@jean-edouard Maybe you can leave a comment that explains how does the TPM functionality being checked

Reply:
This is supposed to be the comment that explains how TPM PCR extension works, I guess it's too cryptic :)
Want me to add more verbiage?

Reply:
@jean-edouard Yes I prefer to remove the cryptic comment. No need to add more comments.